core/vm/instructions.go

```diff
@@ -124,21 +124,8 @@ func opSmod(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory
 
 func opExp(pc *uint64, interpreter *EVMInterpreter, contract *Contract, memory *Memory, stack *Stack) ([]byte, error) {
 	base, exponent := stack.pop(), stack.pop()
-	// some shortcuts
-	cmpToOne := exponent.Cmp(big1)
-	if cmpToOne < 0 { // Exponent is zero
-		// x ^ 0 == 1
-		stack.push(base.SetUint64(1))
-	} else if base.Sign() == 0 {
-		// 0 ^ y, if y != 0, == 0
-		stack.push(base.SetUint64(0))
-	} else if cmpToOne == 0 { // Exponent is one
-		// x ^ 1 == x
-		stack.push(base)
-	} else {
-		stack.push(math.Exp(base, exponent))
-		interpreter.intPool.put(base)
-	}
+	stack.push(math.Exp(base, exponent))
+	interpreter.intPool.put(base)
```

Comment:
This is not correct because `math.Exp` now can return `base` and simultaneously put it both on the stack as well as into the int pool.

Reply:
`math.Exp` is used only by the VM afaik. If so, why do you think it's good to accept a 30% hit?

Reply:
I can agree that the code is cleaner with this, but it's quite a small thing and I would rather keep it the way it is than change it

Reply:
Is 30% hit on an already specific scenario (exp = 0, base = 0 or exp = 1), to keep the code in opExp cleaner is accepable imo. This 30% Is about 70ns slower. Comparing to 2 orders of magnitude speed up with checking the above cases.
What if the base = 1? Should that be also in opExp?